src/StateSmithTest/Output/Sim/diagrams/BeadSorter.sim.html

```diff
@@ -3159,8 +3159,21 @@
         // choose to implement a tracer for debugging purposes.
         sm.tracer = {
             enterState: (mermaidName) => {
-                document.querySelector('g[data-id=' + mermaidName + ']')?.classList.add('active');
-                sm.tracer.log("➡️ Entered " + mermaidName);
+                var e = document.querySelector('g[data-id=' + mermaidName + ']');
+                e?.classList.add('active');
+                var bounds = e?.getBoundingClientRect();
+                if(bounds?.x<0 || bounds?.y<0) {
+                    var x = Math.max(0, -bounds.x + 20);
+                    var y = Math.max(0, -bounds.y + 20);
+                    window.panZoom.panBy({x: x, y: y});
+                }
+                var panebounds = document.querySelector('svg').getBoundingClientRect();
+                if(bounds?.x>panebounds.width || bounds?.y>panebounds.height) {
+                    var x = Math.min(0, panebounds.width-bounds.x-bounds.width-20);
+                    var y = Math.min(0, panebounds.height-bounds.y-bounds.height-20);
```

Comment:
Would you be open to adding some spaces around operators? I tend to go with the standard vscode auto formatter.
```js
var x = Math.min(0, panebounds.width - bounds.x - bounds.width - 20);
var y = Math.min(0, panebounds.height - bounds.y - bounds.height - 20);
```

I guess the auto formatter doesn't help us as we are writing this code in a C# strings.

At some point, I think we will want to break out of writing js inside of C# strings. Maybe have the template js code actually be in a js file so that vscode can analyze/format it? I'd be happy to help with this now. Writing code inside of C# strings isn't very ergonomic.

Further future, we have multiple files js/web files that get packed together into our single `.sim.html` file.

Thoughts?